constraint-solver/src/indexed_constraint_system.rs

```diff
@@ -471,9 +547,11 @@ where
 
     /// Removes the next item from the queue and returns it.
     pub fn pop_front<'a>(&'a mut self) -> Option<ConstraintRef<'a, T, V>> {
-        self.queue
-            .pop_front()
-            .map(|item| item.to_constraint_ref(&self.constraint_system.constraint_system))
+        self.queue.pop_front().map(|item| {
+            item.try_to_constraint_ref(&self.constraint_system.constraint_system)
+                // Derived variables should never be in the queue.
+                .unwrap()
```

Comment:
I'm starting to question the value of having the derived variables in the same pool as constraints and bus interactions, since we have a to treat it as an edge case in a lot of places. What's the tradeoff here?

Reply:
Indeed, they are different, but most of the operations we apply on constraints we also have to apply on the derived variables. We can pull them out, but then the place where we have to treat them as edge cases will just move. I could try to remove them from the queue, but then I would need to introduce a new data structure just for that.

Reply:
I would personally be more in favour of a new data structure, but won't block if you think it's better like this.